File: __tests__/features/auth/loginScreenData.test.ts
import { projectLogin } from '@/features/auth/ui/loginScreenData';

describe('projectLogin', () => {
  it('phone entry: send enabled only for 10 digits', () => {
    expect(projectLogin({ kind: 'unauthenticated' }, '', '')).toEqual({ kind: 'phoneEntry', canSend: false, sending: false });
    expect(projectLogin({ kind: 'unauthenticated' }, '9876543210', '')).toEqual({ kind: 'phoneEntry', canSend: true, sending: false });
    expect(projectLogin({ kind: 'unauthenticated' }, '98765', '')).toEqual({ kind: 'phoneEntry', canSend: false, sending: false });
  });
  it('sendingOtp shows phone entry sending', () => {
    expect(projectLogin({ kind: 'sendingOtp', phone: '9876543210' }, '9876543210', '')).toEqual({ kind: 'phoneEntry', canSend: false, sending: true });
  });
  it('otpSent shows otp entry; verify enabled at 6 digits', () => {
    expect(projectLogin({ kind: 'otpSent', phone: '9876543210' }, '9876543210', '12')).toEqual({ kind: 'otpEntry', phone: '9876543210', canVerify: false, verifying: false });
    expect(projectLogin({ kind: 'otpSent', phone: '9876543210' }, '9876543210', '123456')).toEqual({ kind: 'otpEntry', phone: '9876543210', canVerify: true, verifying: false });
  });
  it('verifying shows otp entry verifying', () => {
    expect(projectLogin({ kind: 'verifying', phone: '9876543210' }, '9876543210', '123456')).toEqual({ kind: 'otpEntry', phone: '9876543210', canVerify: false, verifying: true });
  });
});

File: test/features/auth/login_screen_data_test.dart
import 'package:billpayments/core/mock/mock_network.dart';
import 'package:billpayments/features/auth/data/auth_provider.dart';
import 'package:billpayments/features/auth/ui/login_inputs_provider.dart';
import 'package:billpayments/features/auth/ui/login_screen_data.dart';
import 'package:billpayments/features/auth/ui/login_screen_data_provider.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';
import 'package:flutter_test/flutter_test.dart';

ProviderContainer instantContainer() {
  final container = ProviderContainer(overrides: [
    mockNetworkProvider
        .overrideWithValue(MockNetwork(minDelayMs: 0, maxDelayMs: 1)),
  ]);
  addTearDown(container.dispose);
  return container;
}

void main() {
  test('initial state is phone entry with send disabled', () {
    final container = instantContainer();
    expect(
      container.read(loginScreenDataProvider),
      const LoginScreenData.phoneEntry(canSend: false, sending: false),
    );
  });

  test('ten digit phone enables send', () {
    final container = instantContainer();
    final sub = container.listen(loginScreenDataProvider, (_, _) {});
    container.read(loginPhoneProvider.notifier).edit('9876543210');
    expect(
      container.read(loginScreenDataProvider),
      const LoginScreenData.phoneEntry(canSend: true, sending: false),
    );
    sub.close();
  });

  test('otpSent projects otp entry and six digits enable verify', () async {
    final container = instantContainer();
    final sub = container.listen(loginScreenDataProvider, (_, _) {});
    await container.read(authProvider.notifier).sendOtp('9876543210');
    expect(
      container.read(loginScreenDataProvider),
      const LoginScreenData.otpEntry(
          phone: '9876543210', canVerify: false, verifying: false),
    );
    container.read(loginOtpProvider.notifier).edit('123456');
    expect(
      container.read(loginScreenDataProvider),
      const LoginScreenData.otpEntry(
          phone: '9876543210', canVerify: true, verifying: false),
    );
    sub.close();
  });
}
